pytorch/training/docker/1.7.0/py3/cu101/Dockerfile.gpu

```diff
@@ -0,0 +1,185 @@
+
+RUN wget https://developer.download.nvidia.com/compute/cuda/repos/ubuntu1604/x86_64/libcublas10_${CUBLAS_VERSION}.deb \
+ && dpkg -i libcublas10_${CUBLAS_VERSION}.deb \
+ && apt-get install -f -y \
+ && rm libcublas10_${CUBLAS_VERSION}.deb
+
+RUN wget https://www.open-mpi.org/software/ompi/v4.0/downloads/openmpi-$OPEN_MPI_VERSION.tar.gz \
+ && gunzip -c openmpi-$OPEN_MPI_VERSION.tar.gz | tar xf - \
+ && cd openmpi-$OPEN_MPI_VERSION \
+ && ./configure --prefix=/home/.openmpi \
+ && make all install \
+ && cd .. \
+ && rm openmpi-$OPEN_MPI_VERSION.tar.gz \
+ && rm -rf openmpi-$OPEN_MPI_VERSION
+
+ENV PATH="$PATH:/home/.openmpi/bin"
+ENV LD_LIBRARY_PATH="$LD_LIBRARY_PATH:/home/.openmpi/lib/"
+
+RUN ompi_info --parsable --all | grep mpi_built_with_cuda_support:value \
+ && curl -L -o ~/miniconda.sh https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh \
+ && chmod +x ~/miniconda.sh \
+ && ~/miniconda.sh -b -p /opt/conda \
+ && rm ~/miniconda.sh \
+ && /opt/conda/bin/conda install -y -c anaconda \
+    python=$PYTHON_VERSION \
+    numpy \
+    ipython \
+    mkl \
+    mkl-include \
+    cython \
+    typing \
+    future \
+    "pyopenssl>=17.5.0" \
+ && conda install -c dglteam -y dgl-cuda10.1==0.4.3 \
+ && /opt/conda/bin/conda clean -ya
+
+RUN conda install -c pytorch magma-cuda101==2.5.2 \
+ && conda install -c conda-forge \
+    opencv \
+ && conda install -y scikit-learn \
+    pandas \
+    h5py \
+    requests \
+    libgcc \
+ && conda clean -ya
+
+WORKDIR /opt/pytorch
+
+# Copy workaround script for incorrect hostname
+COPY changehostname.c /
+COPY start_with_right_hostname.sh /usr/local/bin/start_with_right_hostname.sh
+
+WORKDIR /root
+
+RUN /opt/conda/bin/conda config --set ssl_verify False \
+ && pip install --upgrade pip --trusted-host pypi.org --trusted-host files.pythonhosted.org \
+ && ln -s /opt/conda/bin/pip /usr/local/bin/pip3
+
+# Uninstall torch and torchvision before installing the custom versions from an S3 bucket
+RUN pip install \
+    --no-cache-dir smdebug==${SMDEBUG_VERSION} \
+    "sagemaker<2" \
+    sagemaker-experiments==0.* \
+    --no-cache-dir "sagemaker-pytorch-training<3" \
+    --no-cache-dir fastai==1.0.59 \
+    "awscli<2" \
+    psutil \
+    Pillow \
+    scipy \
+    click \
+ && pip install --no-cache-dir -U ${PT_TRAINING_URL} \
+ && pip uninstall -y torchvision \
+ && pip install --no-deps --no-cache-dir -U \
+     https://torchvision-build.s3.amazonaws.com/1.6.0/gpu/torchvision-0.7.0-cp36-cp36m-linux_x86_64.whl
+
+# Install Horovod
+RUN pip uninstall -y horovod \
+ && ldconfig /usr/local/cuda-10.1/targets/x86_64-linux/lib/stubs \
+ && HOROVOD_GPU_ALLREDUCE=NCCL HOROVOD_CUDA_HOME=/usr/local/cuda-10.1 HOROVOD_WITH_PYTORCH=1 pip install --no-cache-dir horovod==${HOROVOD_VERSION} \
+ && ldconfig
+
+# Install Nvidia Apex
+RUN git clone https://github.com/NVIDIA/apex.git \
+ && cd apex \
+ && pip install -v --no-cache-dir --global-option="--cpp_ext" --global-option="--cuda_ext" ./
```

Comment:
Can you try adding `python -m` to the beginning of this line? 

```
python -m pip install -v --no-cache-dir --global-option="--cpp_ext" --global-option="--cuda_ext" ./
```

Reply:
@choidongyeon Why is that needed

Reply:
This was the workaround that had worked for me when working on a different project, but after an online discussion, @ydaiming decided to check out to a specific version.